carbon-design-system/carbon
# fix(HeaderMenu): fix HeaderMenu with isCurrentPage to have a pointer cursor (#10109)

Closes #10109

Pointer cursor is missing because of the `.cds--header__submenu--current` class being added to the list item instead of the link element. Class's styling `top: 0` is blocking `<a>` to be accessible and showing the right cursor. 

There is multiple ways to fix this. I would suggest to use the `.cds--header__menu-item--current` class for HeaderMenu too. This way we could remove duplicated styling too.

#### Changelog

**Changed**

- Added `.cds--header__menu-item--current` in link element instead of using `.cds--header__submenu--current` in list-item element
- Added `isCurrentPage` prop for one story to easily view the functionality in Storybook
- Updated HeaderMenu tests to fit in these changes

**Removed**

- Removed now unused duplicated styles for `.cds--header__submenu--current`

#### Testing / Reviewing

- Check `UI Shell` story named `Header Base w/ Navigation and Actions` https://deploy-preview-12265--carbon-components-react.netlify.app/?path=/story/components-ui-shell--header-base-w-navigation-and-actions
- Ensure hover over HeaderMenu current page (Link 4) shows a pointer cursor
